assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_histogram/run.py

```diff
@@ -532,18 +574,24 @@ def annotate_batch(iterator):
                     client = mlflow.tracking.MlflowClient()
                     client.log_artifacts(
                         child_run_id, DEFAULT_PROMPTFLOW_PATH, artifact_path=artifact_path)
-            tabular_result = pd.read_json(os.path.join(output_dir.name, "eval_results.jsonl"), lines=True)
+            # convert rows from result to pandas dataframe
+            tabular_result = pd.DataFrame(tabular_result["rows"])
             for passthrough_column, passthrough_values in passthrough_cols.items():
                 tabular_result[passthrough_column] = passthrough_values
             # rename metric columns
             try:
                 for column_name in metrics_list:
                     # set failures to -1
-                    tabular_result[column_name] = pd.to_numeric(tabular_result[column_name], errors='coerce')
-                    tabular_result[column_name].fillna(-1, inplace=True)
-                    tabular_result.rename(
-                        columns={column_name: COLUMN_TO_COMPACT_METRIC_NAME[column_name]},
-                        inplace=True)
+                    metric_name_compact = COLUMN_TO_COMPACT_METRIC_NAME[column_name]
+                    # output column names follow schema like "outputs.coherence.gpt_coherence"
+                    result_name = ("outputs." + metric_name_compact + "." + column_name).lower()
+                    tabular_result[result_name] = pd.to_numeric(tabular_result[result_name], errors='coerce')
```

Comment:
any particular reason for choosing coerce or ignore or some other option?

Reply:
yes, that was specifically added in this PR:
https://github.com/Azure/azureml-assets/pull/2382
it's to handle failure cases where promptflow will return (Failed) instead of a numeric value